worker/pkg/workflows/datasync/activities/post-table-sync/activity.go

```diff
@@ -0,0 +1,133 @@
+	connclient mgmtv1alpha1connect.ConnectionServiceClient,
+) *Activity {
+	return &Activity{
+		jobclient:        jobclient,
+		sqlmanagerclient: sqlmanagerclient,
+		connclient:       connclient,
+	}
+}
+
+type RunPostTableSyncRequest struct {
+	JobId string
+	// Identifier that is used in combination with the AccountId to retrieve the post table run config
+	Name      string
+	AccountId string
+}
+type RunPostTableSyncResponse struct {
+}
+
+func (a *Activity) RunPostTableSync(
+	ctx context.Context,
+	req *RunPostTableSyncRequest,
+) (*RunPostTableSyncResponse, error) {
+	activityInfo := activity.GetInfo(ctx)
+	loggerKeyVals := []any{
+		"jobId", req.JobId,
+		"WorkflowID", activityInfo.WorkflowExecution.ID,
+		"RunID", activityInfo.WorkflowExecution.RunID,
+	}
+	logger := log.With(
+		activity.GetLogger(ctx),
+		loggerKeyVals...,
+	)
+	logger.Debug("running post table sync activity")
+	slogger := neosynclogger.NewJsonSLogger().With(loggerKeyVals...)
+
+	jobResp, err := a.jobclient.GetJob(ctx, connect.NewRequest(&mgmtv1alpha1.GetJobRequest{Id: req.JobId}))
+	if err != nil {
+		return nil, fmt.Errorf("unable to get job by id: %w", err)
+	}
+	job := jobResp.Msg.GetJob()
+
+	switch job.Source.Options.Config.(type) {
+	case *mgmtv1alpha1.JobSourceOptions_Postgres, *mgmtv1alpha1.JobSourceOptions_Mysql, *mgmtv1alpha1.JobSourceOptions_Mssql:
+		err = a.runSqlPostTableSync(ctx, req, slogger, job, activityInfo.WorkflowExecution.ID)
+		if err != nil {
+			return nil, err
+		}
+	default:
+	}
+
+	return &RunPostTableSyncResponse{}, nil
+}
+
+func (a *Activity) runSqlPostTableSync(
+	ctx context.Context,
+	req *RunPostTableSyncRequest,
+	slogger *slog.Logger,
+	job *mgmtv1alpha1.Job,
+	workflowId string,
+) error {
+	rcResp, err := a.jobclient.GetRunContext(ctx, connect.NewRequest(&mgmtv1alpha1.GetRunContextRequest{
+		Id: &mgmtv1alpha1.RunContextKey{
+			JobRunId:   workflowId,
+			ExternalId: shared.GetPostTableSyncConfigExternalId(req.Name),
+			AccountId:  req.AccountId,
+		},
+	}))
+	if err != nil {
+		return fmt.Errorf("unable to retrieve posttablesync runcontext for %s: %w", req.Name, err)
+	}
+
+	configBits := rcResp.Msg.GetValue()
+	if len(configBits) == 0 {
+		return nil
+	}
+
+	var config *shared.SqlPostTableSyncConfig
+	err = json.Unmarshal(configBits, &config)
+	if err != nil {
+		return err
+	}
+	if len(config.DestinationStatements) == 0 {
+		return nil
+	}
+
+	for _, destination := range job.Destinations {
+		destinationConnection, err := shared.GetConnectionById(ctx, a.connclient, destination.ConnectionId)
+		if err != nil {
+			return fmt.Errorf("unable to get destination connection (%s) by id: %w", destination.ConnectionId, err)
+		}
+		destDb, err := a.sqlmanagerclient.NewPooledSqlDb(ctx, slogger, destinationConnection)
+		if err != nil {
+			destDb.Db.Close()
+			slogger.Error("unable to connection to destination", "connectionId", destination.ConnectionId)
+			continue
+		}
+		for _, stmt := range config.DestinationStatements {
```

Comment:
I'm a little confused by this logic here.
We iterate over the list of job destination connections - and we invoke the same set of destination statements for each one?

I guess the implicit assumption here is the following:
1. we only make it this far for SQL-enabled connections
2. each sql-enabled destination connection is of the same type (all PG, all MSSQL, all MYSQL) -- which is ok...because that is what we support today. no cross over.

I just read this code and it's not obvious, unless you know that to be true about Neosync, which is maybe the case in many places in our codebase, yet here I am pointing it out for this particular snippet.